src/ksm/client/discovery.go

```diff
@@ -160,7 +172,7 @@ func (sd *discoverer) apiDiscover() (url.URL, error) {
 		if service.Spec.ClusterIP != "" && len(service.Spec.Ports) > 0 {
 			// Look for a port called "http-metrics"
 			for _, port := range service.Spec.Ports {
-				if port.Name == ksmPortName {
+				if port.Name == ksmPortNames[0] {
```

Comment:
shouldn't also look for the others posible names in `ksmPortNames` ?

Reply:
I think that would be ideal, but since we also have:

```
// If not found, return the first TCP port
```

I thought it was not needed to complicate the logic with a for there. Can still be added though.